main.js

```diff
@@ -0,0 +1,126 @@
+const Player = (name, shape) => {
+  const selectedCells = [];
+  const setName = newName => (name = newName);
+  return { name, shape, selectedCells };
+};
+
+class Gameboard {
+  constructor() {
+    this.player1 = Player("Player 1", "X");
+    this.player2 = Player("Player 2", "O");
+    this.currentPlayer = this.player1;
+    this.board = ["", "", "", "", "", "", "", "", ""];
+    this.gameOver = false
+  }
+
+  swapTurn() {
+    if (this.currentPlayer == this.player1) {
+      return (this.currentPlayer = this.player2);
+    } else {
+      //  instructions.innerHTML = `Its's your turn ${this.currentPlayer.name}, your shape is ${this.currentPlayer.shape}`
+      return (this.currentPlayer = this.player1);
+    }
+  }
+
+  gamePlay(shape, index) {
+    if (!this.gameOver) {
+    if (!this.board[index]) {
+      this.board[index] = shape;
+      board.children[index].innerHTML = shape;
+      this.currentPlayer.selectedCells.push(index);
+      if (this.endGame() == this.currentPlayer.name) {
+        return instructions.innerHTML = `Congratulations!!! ${this.currentPlayer.name}, You Win`;
+      } else if (this.endGame() == 'draw') {
+        return instructions.innerHTML = `What a bore, Hit Restart`;
+      }
+      // this.checkWinner(this.endGame())
+      this.swapTurn()
+      instructions.innerHTML = `It's your turn ${this.currentPlayer.name}, your shape is ${this.currentPlayer.shape}`;
+      
+    } else {
+      alert("Select an empty cell");
+    }
+  }
+
+  }
+
+  startNewGame() {
+    return window.location.reload();
+  }
+
+  endGame() {
+    const winningCondition = [
```

Comment:
I don't like this. It is very specific. How can you make it more 'generic'? You could have a function that generates those arrays/patterns

Reply:
Could not figure out this part.